config/configgrpc/configgrpc.go

```diff
@@ -245,7 +246,11 @@ func (gcs *GRPCClientSettings) toDialOptions(host component.Host, settings compo
 		if !valid {
 			return nil, fmt.Errorf("invalid balancer_name: %s", gcs.BalancerName)
 		}
-		opts = append(opts, grpc.WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingPolicy":"%s"}`, gcs.BalancerName)))
+
+		if gcs.BalancerName == "round_robin" {
+			resolver.SetDefaultScheme("dns")
+			opts = append(opts, grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"round_robin":{}}]}`))
+		}
```

Comment:
This will break `BalancerName`s that are not `round_robin`.

Reply:
@TylerHelmuth , thank you for your review.

If you have `balancer_name: foo`, otelcorecol fails to start with the following error.
```
Error: cannot start pipelines: invalid balancer_name: foo
```

If you have `balancer_name:` (empty), or doesn't have the `balancer_name:` node, it defaults to `pick_first`.  Are there other patterns I should test and confirm?

Reply:
I mean to say that previously if `BalancerName` was configured we explicitly set `loadBalancingPolicy` on the service config.  Although `pick_first` may be the default (and currently only other option), I like continuing to explicitly set a value whenever `BalancerName` is set.  This defensive code ensures we are compatible with any `BalancerName` options added in the future.

Reply:
@TylerHelmuth , thank you for the clear explanation. It makes sense. Updated the PR.